akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -544,8 +544,10 @@ private[persistence] trait Eventsourced extends Snapshotter with Stash with Stas
         // while message is in flight, in that case we ignore the call to the handler
         if (id == instanceId) {
           updateLastSequenceNr(p)
+          val applyHandler = applyHandlerOn(p.payload) _
```

Comment:
this is a bit over the top allocation-wise ;-) please call applyHandlerOn directly in the code below